lib/user_name_suggester.rb

```diff
@@ -102,7 +102,9 @@ def self.find_available_username_based_on(name, allowed_username = nil)
   end
 
   def self.fix_username(name)
-    rightsize_username(sanitize_username(name))
+    fixed_username = sanitize_username(name)
+    fixed_username << I18n.t('fallback_username') if fixed_username.empty?
```

Comment:
There might be a problem with the fallback username for non-English locales if the translation contains Unicode characters which aren't allowed (`allowed_unicode_username_characters`) or if `unicode_usernames` is disabled. It might be a good idea to use the English translation in such a case.

Looks like we have a similar problem with the anon usernames.
https://meta.discourse.org/t/how-to-add-anon-before-anonymous-username/185315

Reply:
I've added falling back to _"user"_ in case translation contains disallowed characters. I'm sanitizing the translation firstly and if after sanitizing I've got an empty string I use _"user"_ as a username.

> Looks like we have a similar problem with the anon usernames.
https://meta.discourse.org/t/how-to-add-anon-before-anonymous-username/185315

It looks like the right solution for this is just enabling `unicode_usernames` on the site (just like the user on that topic did). With the solution from this PR, if `unicode_usernames` are disabled, we'll be falling back to _"user"_ for anonymous usernames too. I feel like it's fine. @gschlager what do you think? I can special case resolving anonymous usernames in another PR, so it will be falling back to _"anonymous"_ instead of _"user"_ in case translation contains forbidden characters.

Reply:
Yes, a fallback to "anonymous" would be great.

Reply:
I've addressed it in https://github.com/discourse/discourse/pull/15354.